src/main/java/aquality/selenium/browser/BrowserTabNavigation.java

```diff
@@ -0,0 +1,90 @@
+package aquality.selenium.browser;
+
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+class BrowserTabNavigation implements IBrowserTabNavigation {
+
+    private final RemoteWebDriver driver;
+
+    private void infoLoc(String key, Object... args) {
+        AqualityServices.getLocalizedLogger().info(key, args);
+    }
+
+    BrowserTabNavigation(RemoteWebDriver driver) {
+        this.driver = driver;
+    }
+
+    @Override
+    public String getCurrentTabHandle() {
+        infoLoc("loc.browser.get.tab.handle");
+        return getDriver().getWindowHandle();
+    }
+
+    @Override
+    public Set<String> getTabHandles() {
+        infoLoc("loc.browser.get.tab.handles");
+        return getDriver().getWindowHandles();
+    }
+
+    @Override
+    public void switchToTab(String tabHandle, boolean closeCurrent) {
+        infoLoc("loc.browser.switch.to.tab.handle", tabHandle);
+        closeAndSwitch(tabHandle, closeCurrent);
+    }
+
+    @Override
+    public void switchToTab(int index, boolean closeCurrent) {
+        infoLoc("loc.browser.switch.to.tab.index", index);
+        ArrayList<String> handles = new ArrayList<>(getTabHandles());
```

Comment:
ArrayList<String> handles - use interface instead of implementation - change to List<String>

Reply:
fixed